util/cluster.go

```diff
@@ -0,0 +1,205 @@
+package util
+
+import (
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/columnize"
+	"github.com/giantswarm/gscliauth/config"
+	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/gsctl/client"
+	"github.com/giantswarm/gsctl/client/clienterror"
+	"github.com/giantswarm/gsctl/commands/errors"
+	"github.com/giantswarm/gsctl/confirm"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/afero"
+)
+
+const (
+	listClustersActivityName = "list-clusters"
+	clusterCacheFileName     = "clustercache"
+)
+
+// GetClusterID gets the cluster ID for a provided name/ID
+// by checking in both the user cache and on the API
+func GetClusterID(clusterNameOrID string, clientWrapper *client.Wrapper) (string, error) {
+	isID := IsInClusterCache(clusterNameOrID)
+	if isID {
+		return clusterNameOrID, nil
+	}
+
+	auxParams := clientWrapper.DefaultAuxiliaryParams()
+	auxParams.ActivityName = listClustersActivityName
+
+	response, err := clientWrapper.GetClusters(auxParams)
+	if err != nil {
+		switch {
+		case clienterror.IsUnauthorizedError(err):
+			return "", microerror.Mask(errors.NotAuthorizedError)
+
+		case clienterror.IsAccessForbiddenError(err):
+			return "", microerror.Mask(errors.AccessForbiddenError)
+
+		default:
+			return "", microerror.Mask(err)
+		}
+	}
+
+	var (
+		clusterIDs    []string
+		allClusterIDs []string = make([]string, 0, len(response.Payload))
+	)
+	for _, cluster := range response.Payload {
+		allClusterIDs = append(allClusterIDs, cluster.ID)
+		if matchesValidation(clusterNameOrID, cluster) {
+			clusterIDs = append(clusterIDs, cluster.ID)
+		}
+	}
+
+	if allClusterIDs != nil {
+		CacheClusterIDs(allClusterIDs...)
+	}
+
+	switch {
+	case clusterIDs == nil:
+		return "", microerror.Mask(errors.ClusterNotFoundError)
+
+	case len(clusterIDs) > 1:
+		confirmed, id := handleNameCollision(clusterNameOrID, response.Payload)
+		if !confirmed {
+			return "", nil
+		}
+
+		return id, nil
+
+	default:
+		return clusterIDs[0], nil
+	}
```

Comment:
I think an if / else if construct and a simple return after that would make this easier to read, no?